File: src/main/java/musiccube/entities/UserSongStatus.java
package musiccube.entities;

import javax.persistence.*;

@Entity
public class UserSongStatus {

    @Id
    @GeneratedValue
    @Column
    private int id;

    @ManyToOne
    private Song song;
    @ManyToOne
    private User user;

    @Column
    private boolean listened;
    @Column
    private boolean toListen;

    public int getId() {
        return id;
    }

    public Song getSong() {
        return song;
    }

    public void setSong(Song song) {
        this.song = song;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public boolean isListened() {
        return listened;
    }

    public void setListened(boolean listened) {
        this.listened = listened;
    }

    public boolean isToListen() {
        return toListen;
    }

    public void setToListen(boolean toListen) {
        this.toListen = toListen;
    }
}

File: src/app/Class/UserSongStatus.ts
import {Users} from './Users';
import {Song} from './Song';

export class UserSongStatus {
  id: number;
  song: Song;
  user: Users;
  listened: boolean;
  toListen: boolean;

  constructor(obj?: any) {
    this.id = (obj && obj.id) || 0;
    this.song = (obj && obj.song) || null;
    this.user = (obj && obj.user) || null;
    this.listened = (obj && obj.owned) || false;
    this.toListen = (obj && obj.sought) || false;
  }
}
